### PR TITLE
mesos 1.0.0

### PR DESCRIPTION
**Pull request by haosdent**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
The `inreplace` at `install` time is failing: https://bot.brew.sh/job/Homebrew%20Core%20Pull%20Requests/5983/version=el_capitan/testReport/junit/brew-test-bot/el_capitan/install_mesos/

**Comment by haosdent** [comment]
thx @MikeMcQuaid , just fixed.

**Comment by haosdent** [comment]
Hi, @MikeMcQuaid For mavericks, the build success. For el_capitan and yosemite, it failed because

```
Error: No such dylib name: /usr/local/opt/subversion/lib/libsvn_delta-1.0.dylib
```

However, I have already added

```
  depends_on "subversion"
```

and in my local test, `/usr/local/opt/subversion/lib/libsvn_delta-1.0.dylib` should exists after ` brew install subversion`. Do you have any experience at this? Thank you in advance.


https://bot.brew.sh/job/Homebrew%20Core%20Pull%20Requests/6012/

**Comment by MikeMcQuaid** [comment]
Trying a rebuild.

**Comment by haosdent** [comment]
Hi, @MikeMcQuaid el_capitan and yosemite still failed while I could success on a el_capitan machine.

**Comment by MikeMcQuaid** [comment]
@haosdent Not sure what's going on here but will retry a few times and see if it's a temporary blip. CC @UniqMartin @woodruffw in case this is Ruby MachO related?

**Comment by haosdent** [comment]
Thx @MikeMcQuaid Let me try again

**Comment by haosdent** [comment]
LoL, still failed @MikeMcQuaid

**Comment by UniqMartin** [comment]
Sorry for the trouble! The error is indeed coming from `ruby-macho` and the error message could probably be improved (it currently doesn't state what file it tried to manipulate). I'll take a look and will report back as soon as possible.

**Comment by haosdent** [comment]
@uniqmartin Thanks a lot!

**Comment by UniqMartin** [comment]
@haosdent It might take me a bit longer, as I'm testing in a non-standard Homebrew installation (outside of `/usr/local`) and the build is breaking in this case. If you could investigate this, that would help tremendously. Here's the relevant part from the log:

```
```

**Comment by haosdent** [comment]
Hi, @UniqMartin Thank you for your help. May you told me how to reproduce? So that I could test and verify it local.

**Comment by UniqMartin** [comment]
> May you told me how to reproduce?

If you feel comfortable doing that, you could follow one of the [alternative installation instructions](https://github.com/Homebrew/brew/blob/master/share/doc/homebrew/Installation.md#alternative-installs) and install a second Homebrew in a different location. You should be able to reproduce my issue by trying to build your updated `mesos` formula inside that second Homebrew installation.

**Comment by haosdent** [comment]
Thx @UniqMartin Let me try

**Comment by UniqMartin** [comment]
@haosdent I managed to fix the build for a non-`/usr/local` installation and was now able to reproduce the problem locally (and to investigate it further). This is what I had to add to make it work, though I'm not sure this is the best possible solution:

```diff
diff --git i/Formula/mesos.rb w/Formula/mesos.rb
index 76dc53d3..93cedf46 100644
--- i/Formula/mesos.rb
+++ w/Formula/mesos.rb
@@ -114,6 +114,10 @@ class Mesos < Formula
     system "make"
     system "make", "install"
 
+    # The native Python modules `executor` and `scheduler` (see below) fail to
+    # link to Subversion libraries if Homebrew isn't installed in `/usr/local`.
+    ENV.append_to_cflags "-L#{Formula["subversion"].opt_lib}"
+
     system "./configure", "--enable-python", *args
     ["native", "interface", "executor", "scheduler", "cli", ""].each do |p|
       cd "src/python/#{p}" do
```

Would be good to add this or something with a similar effect to the formula, so that it can be built by other users that don't use `/usr/local` for their Hombrew installations.

I have also improved the error handling for the error that you've seen, thus the output should be a lot less cryptic (this should also appear in the CI log once the rescheduled build finishes):

```console
$ brew bottle --verbose --json mesos
==> Determining mesos bottle revision...
==> Bottling mesos-1.0.0.el_capitan.bottle.tar.gz...
Error: Failed changing install name in /opt/brewery/tests/Cellar/mesos/1.0.0/lib/python2.7/site-packages/mesos/executor/_executor.so
  from /opt/brewery/tests/opt/subversion/lib/libsvn_delta-1.0.dylib
    to @@HOMEBREW_PREFIX@@/opt/subversion/lib/libsvn_delta-1.0.dylib
Error: No such dylib name: /opt/brewery/tests/opt/subversion/lib/libsvn_delta-1.0.dylib
```

The underlying issue is that two files built by this formula are built as both 32- and 64-bit libraries, though the Subversion they are linking against only has 64-bit libraries. Consequently, the 32-bit part links to a different set of libraries and this is confusing our Mach-O handling. This seems to be a variation of the problem we've encountered recently with our Python `virtualenv` support and that was addressed in Homebrew/brew#600.

@tdsmith Could you have a look at this and provide some of your Python wisdom? I'm not sure how to prevent this behavior in this instance. Maybe relevant in this case is that the formula is built using `stdenv` because it depends on `subversion` and that depends on `scons` which ends up disabling `superenv`. Here's the `otool -arch all -L` output for the problematic files:

```console
$ otool -arch all -L lib/python2.7/site-packages/mesos/executor/_executor.so
lib/python2.7/site-packages/mesos/executor/_executor.so (architecture i386):
	/usr/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.5)
	/usr/lib/libsasl2.2.dylib (compatibility version 3.0.0, current version 3.15.0)
	/usr/lib/libcurl.4.dylib (compatibility version 7.0.0, current version 8.0.0)
	/usr/lib/libapr-1.0.dylib (compatibility version 5.0.0, current version 5.8.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 120.1.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1226.10.1)
lib/python2.7/site-packages/mesos/executor/_executor.so (architecture x86_64):
	/usr/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.5)
	/opt/brewery/tests/opt/subversion/lib/libsvn_delta-1.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	/opt/brewery/tests/opt/subversion/lib/libsvn_subr-1.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	/usr/lib/libsasl2.2.dylib (compatibility version 3.0.0, current version 3.15.0)
	/usr/lib/libcurl.4.dylib (compatibility version 7.0.0, current version 8.0.0)
	/usr/lib/libapr-1.0.dylib (compatibility version 5.0.0, current version 5.8.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 120.1.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1226.10.1)

$ otool -arch all -L lib/python2.7/site-packages/mesos/scheduler/_scheduler.so
lib/python2.7/site-packages/mesos/scheduler/_scheduler.so (architecture i386):
	/usr/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.5)
	/usr/lib/libsasl2.2.dylib (compatibility version 3.0.0, current version 3.15.0)
	/usr/lib/libcurl.4.dylib (compatibility version 7.0.0, current version 8.0.0)
	/usr/lib/libapr-1.0.dylib (compatibility version 5.0.0, current version 5.8.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 120.1.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1226.10.1)
lib/python2.7/site-packages/mesos/scheduler/_scheduler.so (architecture x86_64):
	/usr/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.5)
	/opt/brewery/tests/opt/subversion/lib/libsvn_delta-1.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	/opt/brewery/tests/opt/subversion/lib/libsvn_subr-1.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	/usr/lib/libsasl2.2.dylib (compatibility version 3.0.0, current version 3.15.0)
	/usr/lib/libcurl.4.dylib (compatibility version 7.0.0, current version 8.0.0)
	/usr/lib/libapr-1.0.dylib (compatibility version 5.0.0, current version 5.8.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 120.1.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1226.10.1)
```

**Comment by haosdent** [comment]
Hi, @UniqMartin Thanks a lot for your kindly help. Mesos only support 64 bits. I think we could ignore 32 bits.

**Comment by UniqMartin** [comment]
> Mesos only support 64 bits. I think we could ignore 32 bits.

I was imagining this would be the case and all other components are building only 64 bit binaries, which confirms this to a degree. I think this is a build system failure or some unfortunate interaction with the system Python on macOS (that is still a universal binary that supports both 32 and 64 bits). If you know how to fix this, please do so. Otherwise @tdsmith will hopefully be able to help soon. (I'm unfortunately of no use when dealing with Python.)

**Comment by haosdent** [comment]
Thx @UniqMartin In my machine, seems only could find 64 bits.
```
./src/python/executor/build/lib.macosx-10.10-x86_64-2.7/mesos/executor/_executor.so:
        /usr/local/opt/gflags/lib/libgflags.2.dylib (compatibility version 2.0.0, current version 2.1.2)
        /usr/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.5)
        /usr/local/opt/subversion/lib/libsvn_delta-1.0.dylib (compatibility version 1.0.0, current version 1.0.0)
        /usr/local/opt/subversion/lib/libsvn_subr-1.0.dylib (compatibility version 1.0.0, current version 1.0.0)
        /usr/lib/libsasl2.2.dylib (compatibility version 3.0.0, current version 3.15.0)
        /usr/lib/libcurl.4.dylib (compatibility version 7.0.0, current version 8.0.0)
        /usr/lib/libapr-1.0.dylib (compatibility version 5.0.0, current version 5.8.0)
        /usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 120.1.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1226.10.1)
```

**Comment by tdsmith** [comment]
>  I think this is a build system failure or some unfortunate interaction with the system Python on macOS (that is still a universal binary that supports both 32 and 64 bits).

Yes; the only way I know to convince a universal Python not to try to build for all arches is to strip arch flags with superenv.

**Comment by UniqMartin** [comment]
> In my machine, seems only could find 64 bits.

@haosdent I guess you have a non-system Python installed, most likely the Homebrew `python` formula? You can check by inspecting the output of `brew config` that will report `/usr/bin/python` in the “Python:” line if you have system Python and something else otherwise. Since the Homebrew Python is a single-architecture build, you will get a different result with it.

> Yes; the only way I know to convince a universal Python not to try to build for all arches is to strip arch flags with superenv.

@tdsmith Thanks for your input! Not exactly the answer I was hoping for. Should we just move ahead and ignore the issue, merge Homebrew/brew#656 as-is (like `install_name_tool` it will be more tolerant of such discrepancies) or do you see a feasible way this could be fixed? Ideally, it would be nice to fix this, but the (dysfunctional) 32-bit part of the Python extensions isn't causing any harm aside from some wasted disk space and is probably never loaded. Thoughts?

**Comment by haosdent** [comment]
Hi, @UniqMartin Yes, I use the python installed by homebrew.

**Comment by UniqMartin** [comment]
I rescheduled a new build now that the (somewhat relaxed) Mach-O related changes have landed in Homebrew. I expect the build to finish just fine this time. If it does, this can probably be merged. (Just saying because I won't be really around the next few days to do it myself.)

**Comment by haosdent** [comment]
Thx @UniqMartin It looks we could merge it now?

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew and patience @haosdent! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format for simple version updates is `mesos 1.0.0`. Please complete the issue template in future PRs where this format is detailed in the linked [Contributing guidelines](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md). Thanks!

**Comment by haosdent** [comment]
Thanks a lot for your kindly helps @MikeMcQuaid @UniqMartin @tdsmith
